solutionbox/structured_data/test_mltoolbox/test_sd_trainer.py

```diff
@@ -101,21 +115,27 @@ def _check_training_screen_output(self, accuracy=None, loss=None):
       accuracy: float. Eval accuracy should be > than this number.
       loss: flaot. Eval loss should be < than this number.
     """
-    # Print the last line of training output which has the loss value.
+    # Find the last training loss line in the output
     lines = self._training_screen_output.splitlines()
     last_line = None
     for line in lines:
       if line.startswith('INFO:tensorflow:Loss for final step:'):
         print(line)
-      if line.startswith('INFO:tensorflow:Saving dict for global step 2500'):
+      if line.startswith('INFO:tensorflow:Saving dict for global step %d' 
+                          % self._max_steps):
         last_line = line
         break
+
     if not last_line:
       print('Skipping _check_training_screen_output as could not find last eval'
             ' line')
       return
     print(last_line)
 
+    if not self._check_model_loss:
```

Comment:
_check_model_loss seems misleading since you also check something else if enabled, such as global step. Also the function takes "loss" argument, and if it is None it does not check loss. So there is already a flag for that?